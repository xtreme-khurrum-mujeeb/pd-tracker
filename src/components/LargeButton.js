import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity
} from 'react-native';

const LargeButton = ({ name }) => {
  return (
    <TouchableOpacity activeOpacity={0.5}>
      <View style={styles.button}>
        <Text style={styles.buttonText}> {name}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#FFF',
    paddingVertical: 8,
    marginVertical: 10,
    borderRadius: 5,
    alignItems: 'center',
  },
  buttonText: {
    color: '#008774',
    fontFamily: 'Avenir Next',
    fontSize: 20
  }
});

export default LargeButton;
