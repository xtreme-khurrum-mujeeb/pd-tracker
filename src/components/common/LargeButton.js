import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity
} from 'react-native';

import styles from '../../styles';
import colors from '../../colors';


const LargeButton = ({ onPress, name }) => {
  return (
    <TouchableOpacity onPress={onPress} activeOpacity={0.5}>
      <View style={localStyles.button}>
        <Text style={[localStyles.buttonText, styles.h3]}> {name}</Text>
      </View>
    </TouchableOpacity>
  );
};

const localStyles = StyleSheet.create({
  button: {
    backgroundColor: colors.pure_white,
    paddingVertical: 8,
    marginVertical: 10,
    borderRadius: 5,
    alignItems: 'center',
  },
  buttonText: {
    color: colors.text_green,
  }
});

export { LargeButton };
