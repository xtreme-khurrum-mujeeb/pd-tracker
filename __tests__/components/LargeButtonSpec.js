import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LargeButton from '../../src/components/LargeButton';

describe('LargeButton', () => {
  let onPress;
  let initialState;

  beforeEach(() => {
    onPress = jest.fn();
    initialState = {
      onPress,
      name: 'some text',
    };
  });

  it('renders the text', () => {
    const tree = renderer.create(
      <LargeButton {...initialState} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress when clicking the button', () => {
    let subject = shallow(<LargeButton {...initialState} />);
    subject.find('TouchableOpacity').simulate('press');
    expect(onPress.mock.calls.length).toEqual(1);
  });
});
