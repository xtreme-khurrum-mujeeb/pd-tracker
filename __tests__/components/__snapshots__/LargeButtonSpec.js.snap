// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeButton renders the text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "marginVertical": 10,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#008774",
          "fontFamily": "Avenir Next",
          "fontSize": 20,
        }
      }
    >
       
      some text
    </Text>
  </View>
</View>
`;
